src/main/kotlin/org/jitsi/nlj/rtp/PaddingVideoPacket.kt

```diff
@@ -18,7 +18,10 @@ package org.jitsi.nlj.rtp
 
 import org.jitsi.nlj.util.BufferPool
 
-//TODO: need to set the padding amount at the end of the payload!
 class PaddingVideoPacket(
     length: Int
-) : VideoRtpPacket(BufferPool.getBuffer(length), 0, length)
\ No newline at end of file
+) : VideoRtpPacket(BufferPool.getBuffer(length), 0, length) {
+    init {
+        paddingSize = payloadLength
```

Comment:
Getting the payload length here calculates the header length, which is based on whatever the byte[] from the pool used to have. Or am I missing something?

Reply:
nope, you're exactly right :) ran into this last night, will push the fix here.